src/models/position/batch.js

```diff
@@ -211,6 +211,63 @@ const getTransferByIdsForReserve = async (trx, transferIds) => {
   return {}
 }
 
+const getFxTransferInfoList = async (trx, commitRequestId, transferParticipantRoleTypeId, ledgerEntryTypeId) => {
+  try {
+    const knex = await Db.getKnex()
+    const transferInfos = await knex('fxTransferParticipant')
+      .transacting(trx)
+      .where({
+        'fxTransferParticipant.transferParticipantRoleTypeId': transferParticipantRoleTypeId,
+        'fxTransferParticipant.ledgerEntryTypeId': ledgerEntryTypeId
+      })
+      .whereIn('fxTransferParticipant.commitRequestId', commitRequestId)
+      .select(
+        'fxTransferParticipant.*'
+      )
+    const info = {}
+    // This should key the transfer info with the latest transferStateChangeId
+    for (const transferInfo of transferInfos) {
+      if (!(transferInfo.commitRequestId in info)) {
+        info[transferInfo.commitRequestId] = transferInfo
+      }
+    }
+    return info
+  } catch (err) {
+    Logger.isErrorEnabled && Logger.error(err)
+    throw err
+  }
+}
+
+const getReservedPositionChangesByCommitRequestIds = async (trx, commitRequestIdList) => {
+  try {
+    const knex = await Db.getKnex()
+    const participantPositionChanges = await knex('fxTransferStateChange')
+      .transacting(trx)
+      .whereIn('fxTransferStateChange.commitRequestId', commitRequestIdList)
+      .where('fxTransferStateChange.transferStateId', Enum.Transfers.TransferInternalState.RESERVED)
+      .leftJoin('participantPositionChange AS ppc', 'ppc.fxTransferStateChangeId', 'fxTransferStateChange.fxTransferStateChangeId')
+      .leftJoin('participantPosition AS pp', 'pp.participantPositionId', 'ppc.participantPositionId')
+      .select(
+        'ppc.*',
+        'fxTransferStateChange.commitRequestId AS commitRequestId',
+        'pp.participantCurrencyId AS participantCurrencyId'
+      )
+    const info = {}
+    for (const participantPositionChange of participantPositionChanges) {
+      if (!(participantPositionChange.commitRequestId in info)) {
+        info[participantPositionChange.commitRequestId] = {}
+      }
+      if (participantPositionChange.participantCurrencyId) {
+        info[participantPositionChange.commitRequestId][participantPositionChange.participantCurrencyId] = participantPositionChange
```

Comment:
This logic seems to work if we assume there will be only one participantPositionChange per participantCurrencyId for a given commitRequestId. Although we don't have any use case to have multiple, but is it safe to assume so in the model level?

Reply:
Yes it is assuming that there is 1 RESERVED type participant position change per commitRequestId per account.
Yea...it doesn't feel right. Where do you propose we validate such assumptions?

Reply:
I think, lets leave that assumption as an elaborated comment for now.

Reply:
done. added a comment on the model function.